sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -591,10 +603,7 @@ ssize_t coap_build_reply_header(coap_pkt_t *pkt, unsigned code,
                 payload = NULL;
             }
 
-            /* no-response requested, only send empty ACK or nothing */
-            if (type != COAP_TYPE_ACK) {
-                return 0;
-            }
+            return -ECANCELED;
```

Comment:
The documentation of `coap_build_reply_header()` (and `coap_reply_simple()` which makes use of this) will now also need an update.

Reply:
Done and rebased to fix the merge conflicts